src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs

```diff
@@ -29,11 +31,11 @@ internal unsafe class RaspberryPi3LinuxDriver : GpioDriver
         private const string DeviceTreeRanges = "/proc/device-tree/soc/ranges";
         private const string ModelFilePath = "/proc/device-tree/model";
 
-        private readonly PinState[] _pinModes;
+        private readonly PinState?[] _pinModes;

```

Comment:
Why is this nullable? seems like we can switch it to non-nullable given we always instantiate it, and that way we can save some of the exclamation points you have to use bellow.